plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/BlockCrackCommand.java

```diff
@@ -57,62 +58,33 @@ public BlockCrackCommand() {
     // -->
 
-    @Override
-    public void parseArgs(ScriptEntry scriptEntry) throws InvalidArgumentsException {
-        for (Argument arg : scriptEntry) {
-            if (arg.matchesPrefix("players")
-                    && arg.matchesArgumentList(PlayerTag.class)) {
-                scriptEntry.addObject("players", arg.asType(ListTag.class).filter(PlayerTag.class, scriptEntry));
-            }
-            else if (arg.matchesPrefix("progress")
-                    && arg.matchesInteger()) {
-                scriptEntry.addObject("progress", arg.asElement());
-            }
-            else if (arg.matchesArgumentType(LocationTag.class)) {
-                scriptEntry.addObject("location", arg.asType(LocationTag.class));
-            }
-            else if (arg.matches("stack")) {
-                scriptEntry.addObject("stack", new ElementTag(true));
-            }
-            else {
-                arg.reportUnhandled();
-            }
-        }
-        if (!scriptEntry.hasObject("progress")) {
-            throw new InvalidArgumentsException("Must specify crack animation progress!");
-        }
-        if (!scriptEntry.hasObject("location")) {
-            throw new InvalidArgumentsException("Must specify a valid location!");
-        }
-        scriptEntry.defaultObject("players", Collections.singletonList(Utilities.getEntryPlayer(scriptEntry)))
-                .defaultObject("stack", new ElementTag(false));
-    }
-
     private static class IntHolder {
         public int theInt;
         public int base;
     }
 
-    private static Map<Location, Map<UUID, IntHolder>> progressTracker = new HashMap<>();
+    private static final Map<Location, Map<UUID, IntHolder>> progressTracker = new HashMap<>();
     private static int lastBase;
 
-    @Override
-    public void execute(ScriptEntry scriptEntry) {
-        List<PlayerTag> players = (List<PlayerTag>) scriptEntry.getObject("players");
-        ElementTag progress = scriptEntry.getElement("progress");
-        LocationTag location = scriptEntry.getObjectTag("location");
-        ElementTag stack = scriptEntry.getElement("stack");
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), db("players", players), progress, location, stack);
+    public static void autoExecute(ScriptEntry scriptEntry,
+                            @ArgName("location") @ArgLinear LocationTag location,
+                            @ArgName("progress") @ArgPrefixed int progress,
+                            @ArgName("stack") boolean stack,
+                            @ArgName("players") @ArgDefaultNull @ArgPrefixed @ArgSubType(PlayerTag.class) List<PlayerTag> players,
+                            @ArgName("duration") @ArgPrefixed @ArgDefaultNull DurationTag duration) {
+        if (players == null) {
+            players = Collections.singletonList(Utilities.getEntryPlayer(scriptEntry));
         }
-        Location loc = location.getBlock().getLocation();

```

Comment:
This line should not have been removed, it is intentional regularization of the location value to ensure the map works reliably